code/modules/projectiles/projectile.dm

```diff
@@ -743,35 +743,40 @@ So if we are on the 32th absolute pixel coordinate we are on tile 1, but if we a
 
 	var/feedback_flags = NONE
 
-	var/living_armor = (proj.ammo.flags_ammo_behavior & AMMO_IGNORE_ARMOR) ? 0 : get_living_armor(proj.armor_type, proj.def_zone, proj.dir)
-	if(living_armor)
+	var/hard_armor = (proj.ammo.flags_ammo_behavior & AMMO_IGNORE_ARMOR) ? 0 : get_hard_armor(proj.armor_type, proj.def_zone, proj.dir)
+	var/soft_armor = (proj.ammo.flags_ammo_behavior & AMMO_IGNORE_ARMOR) ? 0 : get_soft_armor(proj.armor_type, proj.def_zone, proj.dir)
+	if(hard_armor || soft_armor)
 		var/penetration = proj.ammo.penetration
 		if(penetration > 0)
 			if(proj.shot_from && src == proj.shot_from.sniper_target(src))
 				damage *= SNIPER_LASER_DAMAGE_MULTIPLIER
 				penetration *= SNIPER_LASER_ARMOR_MULTIPLIER
 				add_slowdown(SNIPER_LASER_SLOWDOWN_STACKS)
-			living_armor -= penetration
-		switch(living_armor)
-			if(-INFINITY to 0) //Armor fully penetrated.
-				feedback_flags |= BULLET_FEEDBACK_PEN
-			if(100 to INFINITY) //Damage invulnerability.
+			if(hard_armor)
+				hard_armor = max(0, hard_armor - (hard_armor * penetration * 0.01)) //AP reduces a % of hard armor.
+			if(soft_armor)
+				soft_armor = max(0, soft_armor - penetration) //Flat removal.
+
+		if(!hard_armor && !soft_armor) //Armor fully penetrated.
+			feedback_flags |= BULLET_FEEDBACK_PEN
+		else
+			if(hard_armor)
+				damage = max(0, damage - hard_armor) //Damage soak.
+			if(!damage) //Damage fully soaked.
+				bullet_soak_effect(proj)
+				feedback_flags |= BULLET_FEEDBACK_IMMUNE
+			else if(soft_armor >= 100) //Damage invulnerability.
 				damage = 0
 				bullet_soak_effect(proj)
 				feedback_flags |= BULLET_FEEDBACK_IMMUNE
-			else
-				damage = max(0, damage - (living_armor * 0.1)) //Hard armor, damage soak. 10% of the armor's value.
-				if(!damage) //Damage fully soaked.
-					bullet_soak_effect(proj)
-					feedback_flags |= BULLET_FEEDBACK_SOAK
-				else
-					living_armor *= 0.9 //Remove the 10% that was used to soak damage.
-					damage -= damage * living_armor * 0.01 //Soft armor/padding, damage reduction.
+			else if(soft_armor) //Soft armor/padding, damage reduction.
+				feedback_flags |= BULLET_FEEDBACK_SOAK
+				damage -= damage * soft_armor * 0.01
 
 	if(proj.ammo.flags_ammo_behavior & AMMO_INCENDIARY)
-		living_armor = get_living_armor("fire", proj.def_zone) //This value could potentially be negative, indicating fire weakness.
-		if(living_armor < 100) //If armor is 100 or over the mob is fireproof.
-			adjust_fire_stacks(CEILING(10 - (10 * (living_armor / 100)), 1)) //We could add an ammo fire strength in time, as a variable.
+		soft_armor = get_soft_armor("fire", proj.def_zone) //This value could potentially be negative, indicating fire weakness.
+		if(soft_armor < 100) //If armor is 100 or over the mob is fireproof.
```

Comment:
in your comment on https://github.com/tgstation/TerraGov-Marine-Corps/pull/3931#discussion_r419178802 you talked about using hard armor for fireproof, but here its soft armor.

Do we want to align that?